### PR TITLE
Improve Itinerary and CO2 Types

### PR DESCRIPTION
**Pull request by daniel-heppner-ibigroup**
This PR adds a new type for mass units, and also a "looser" Itinerary type that makes all fields optional except legs, since in most places that is the only property we need. This will make Typescript happier when we are making "fake" itineraries in OTP-RR and in testing.

### CODE REVIEW

File: packages/types/src/index.ts [hunk 1]
```diff
@@ -400,6 +400,13 @@ export type Itinerary = {
   walkTime: number;
 };
 
+/**
+ * In many places all we need from the Itinerary is the legs,
+ * this type makes all the other types optional except legs.
+ */
+export type ItineraryOnlyLegsRequired = Partial<Itinerary> &
```
Comment (miles-grant-ibigroup) [inline packages/types/src/index.ts:407]: super nitpick: can we put this on one line?

**Review by miles-grant-ibigroup — CHANGES_REQUESTED** [review]
Not the hugest fan of the second Itinerary type just for this one use case -- some advanced typescript going on as well. Is there no way to make the itinerary type have optional fields?

**Comment by daniel-heppner-ibigroup** [comment]
What do you mean by one use case? It's used like this everywhere in the itinerary core utils file. I wasn't sure about modifying the Itinerary type to make all the fields permanently optional, it would probably break a lot of other stuff. @miles-grant-ibigroup

**Comment by miles-grant-ibigroup** [comment]
I thought we were making this change to make the tests work with typescript! In that case I'd either rename the new type to make it clear what it's for or try to unify the types.

**Comment by daniel-heppner-ibigroup** [comment]
> I thought we were making this change to make the tests work with typescript! In that case I'd either rename the new type to make it clear what it's for or try to unify the types.

Let me explain what's going on a bit better here; I needed a more relaxed type for the calculateEmissions function, since when there is no driving itinerary in OTP, we generate a fake one that only contains the legs property. Typescript isn't happy about this object not containing any other Itinerary properties, but we don't actually need those. That's what I made this type for. Then I realized that actually, none of these functions require anything besides the legs, so I just applied the type to all of them. 

I can name it something else, but I thought it would be good just name it something super obvious.

**Comment by miles-grant-ibigroup** [comment]
Thanks for that explanation! This all makes sense to me. I don't have a better alternative name right now, so will defer to the second reviewer

File: packages/trip-details/src/types.ts [hunk 2]
```diff
@@ -1,6 +1,6 @@
 // Prettier does not recognize the import type syntax.
 // eslint-disable-next-line prettier/prettier
-import type { FareDetails, Money, Itinerary, Leg, Fare, } from "@opentripplanner/types";
+import type { FareDetails, Money, Itinerary, Leg, Fare, MassUnitOption, } from "@opentripplanner/types";
```
Comment (binh-dam-ibigroup) [inline packages/trip-details/src/types.ts:3]: Sort imports, remove trailing comma

**Review by binh-dam-ibigroup — APPROVED** [review]
LGTM

**Pull request merged**